main_test.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type addTest struct {
+	testCaseName   string
+	ip             string
+	expectedOutput string
+}
```

Comment:
**Suggestion:** In your test cases, you are using a string to represent the IP address. It would be more idiomatic in Go to use a net.IP type instead. This will also allow you to use the built-in functions of the net package to validate the IP addresses.
```suggestion
type addTest struct {
	testCaseName   string
	ip             net.IP
	expectedOutput string
}
```